akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/StashSpec.scala

```diff
@@ -242,3 +249,340 @@ abstract class StashSpec extends ScalaTestWithActorTestKit with WordSpecLike {
   }
 
 }
+
+class UnstashingSpec extends ScalaTestWithActorTestKit("""
+  akka.loggers = ["akka.testkit.TestEventListener"]
+  """) with WordSpecLike {
+
+  // needed for EventFilter
+  private implicit val untypedSys: akka.actor.ActorSystem = {
+    import akka.actor.typed.scaladsl.adapter._
+    system.toUntyped
+  }
+
+  private def slowStoppingChild(latch: CountDownLatch): Behavior[String] =
+    Behaviors.receiveSignal {
+      case (_, PostStop) ⇒
+        latch.await(10, TimeUnit.SECONDS)
+        Behaviors.same
+    }
+
+  private def stashingBehavior(
+    probe:                 ActorRef[String],
+    withSlowStoppingChild: Option[CountDownLatch] = None) = {
+    Behaviors.setup[String] { ctx ⇒
+
+      withSlowStoppingChild.foreach(latch ⇒ ctx.spawnAnonymous(slowStoppingChild(latch)))
+
+      val stash = StashBuffer[String](10)
+
+      def unstashing(n: Int): Behavior[String] =
+        Behaviors.receiveMessage[String] {
+          case "stash" ⇒
+            probe.ref ! s"unstashing-$n"
+            unstashing(n + 1)
+          case "stash-fail" ⇒
+            probe.ref ! s"stash-fail-$n"
+            throw TestException("unstash-fail")
+          case "get-current" ⇒
+            probe.ref ! s"current-$n"
+            Behaviors.same
+          case "get-stash-size" ⇒
+            probe.ref ! s"stash-size-${stash.size}"
+            Behaviors.same
+          case "unstash" ⇒
+            // when testing resume
+            stash.unstashAll(ctx, unstashing(n))
+        }.receiveSignal {
+          case (_, PreRestart) ⇒
+            probe.ref ! s"pre-restart-$n"
+            Behaviors.same
+          case (_, PostStop) ⇒
+            probe.ref ! s"post-stop-$n"
+            Behaviors.same
+        }
+
+      Behaviors.receiveMessage[String] {
+        case msg if msg.startsWith("stash") ⇒
+          stash.stash(msg)
+          Behavior.same
+        case "unstash" ⇒
+          stash.unstashAll(ctx, unstashing(0))
+        case "get-current" ⇒
+          probe.ref ! s"current-00"
+          Behaviors.same
+        case "get-stash-size" ⇒
+          probe.ref ! s"stash-size-${stash.size}"
+          Behaviors.same
+      }
+    }
+  }
+
+  "Unstashing" must {
+
+    "work with initial Behaviors.same" in {
+      // FIXME #26148 unstashAll doesn't support Behavior.same
```

Comment:
this is something we can follow up in a new ticket

Reply:
Thanks for adding the issue number :)